### PR TITLE
Node Exporter 

### PR DESCRIPTION
**Pull request by darkdrgn2k**
Added installation of Node Exporter (part of Prometheus) to script

### CODE REVIEW

File: scripts/grafana/install [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+if $(uname -m | grep -Eq ^armv6); then
+    URL="https://bintray.com/fg2it/deb-rpi-1b/download_file?file_path=main%2Fg%2Fgrafana_4.3.2_armhf.deb"
+fi
+
+sudo apt-get -y install libfontconfig1 fontconfig-config fonts-dejavu-core ttf-bitstream-vera fonts-liberation sqlite3
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Download and install IPFS
```
Comment (benhylau) [inline scripts/grafana/install]: Grafana

File: scripts/grafana/install [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+mkdir "$BASE_DIR/tmp"
+
+wget ${URL} -O "$BASE_DIR/tmp/go-grafana.tar.gz"
+
+sudo dpkg -i "$BASE_DIR/tmp/go-grafana.tar.gz"
+
+rm -rf "$BASE_DIR/tmp"
```
Comment (benhylau) [inline scripts/grafana/install]: Maybe we should change these all the /tmp the way you did the nodejs install

File: scripts/grafana/install [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+sudo /bin/systemctl daemon-reload
+sudo /bin/systemctl enable grafana-server
+sudo /bin/systemctl start grafana-server
+
+sleep 2 #Wait for it to start
```
Comment (benhylau) [inline scripts/grafana/install]: The previous call is blocking right? Is this necessary? It seems not most reliable to simply sleep...

File: scripts/grafana/uninstall [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Uninstall grafana
+
+sudo /bin/systemctl disable grafana-server
```
Comment (benhylau) [inline scripts/grafana/uninstall]: Just `systemctl` like elsewhere?

File: scripts/install2 [hunk 5]
```diff
@@ -58,6 +61,45 @@ if [ -z "$WITH_IPFS" -o "$WITH_IPFS" != "true" -a "$WITH_IPFS" != "false" ]; the
...
+        echo -e "\e[1;32mNode Exporter will be installed on port 9100\e[0m"
+        WITH_NODE_EXPORTER=true
+    fi
+fi
+if [ "$WITH_NODE_EXPORTER"==true ]; then
+    if [ -z "$WITH_PROMETHEUS" -o "$WITH_PROMETHEUS" != "true" -a "$WITH_PROMETHEUS" != "false" ]; then
+        read -p "Install Premetheus (Monitoring) (Y/n)? " -n 1 -r
+        echo ""
+        if [[ $REPLY =~ ^[Nn]$ ]]; then
+            echo -e "\e[1;31mPremetheus installation will be skipped\e[0m"
```
Comment (benhylau) [inline scripts/install2]: Typo

File: scripts/install2 [hunk 6]
```diff
@@ -137,6 +179,21 @@ if [ ! -x "$(command -v ipfs)" ] && [ ! -z "$WITH_IPFS" -a "$WITH_IPFS" == "true
...
+if [ ! -x "$(command -v node_exporter)" ] && [ ! -z "$WITH_NODE_EXPORTER" -a "$WITH_NODE_EXPORTER" == "true" ]; then
+    source node_exporter/install
+fi
+
+# PROMETHEUS
+if [ ! -x "$(command -v /opt/prometheus/prometheus)" ] && [ ! -z "$WITH_PROMETHEUS" -a "$WITH_PROMETHEUS" == "true" ]; then
+    source prometheus/install
+fi
+
+#
```
Comment (benhylau) [inline scripts/install2]: ?

File: scripts/node_exporter/install [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+GO_NODE_EXPORTER_VERSION=0.14.0
+ARM_VERSION=armv7
+if $(uname -m | grep -Eq ^armv6); then
+    ARM_VERSION=armv6
+fi
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Download and install IPFS
+mkdir "$BASE_DIR/tmp"
```
Comment (benhylau) [inline scripts/node_exporter/install]: /tmp ?

File: scripts/node_exporter/install [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+wget "https://github.com/prometheus/node_exporter/releases/download/v${GO_NODE_EXPORTER_VERSION}/node_exporter-${GO_NODE_EXPORTER_VERSION}.linux-${ARM_VERSION}.tar.gz" -O "$BASE_DIR/tmp/go-node_exporter.tar.gz"
+tar xvfz "$BASE_DIR/tmp/go-node_exporter.tar.gz" -C "$BASE_DIR/tmp" --strip 1
+
+sudo cp "$BASE_DIR/tmp/node_exporter" /usr/local/bin/node_exporter
+sudo chown root:staff /usr/local/bin/node_exporter
+rm -rf "$BASE_DIR/tmp"
+
+# Configure systemd to start ipfs-daemon.service on system boot
```
Comment (benhylau) [inline scripts/node_exporter/install]: `ipfs-daemon.service`

File: scripts/node_exporter/install [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+sudo cp "$BASE_DIR/node_exporter-daemon.service" /lib/systemd/system/node_exporter.service
```
Comment (benhylau) [inline scripts/node_exporter/install]: Use `-` as separator for service names

File: scripts/uninstall [hunk 10]
```diff
@@ -29,6 +29,8 @@ source "$BASE_DIR/mesh-point/uninstall"
 source "$BASE_DIR/hostapd/uninstall"
 source "$BASE_DIR/ipfs/uninstall"
 source "$BASE_DIR/extra-tools/uninstall"
+source "$BASE_DIR/node_exporter/uninstall"
+source "$BASE_DIR/prometheus/uninstall"
```
Comment (darkdrgn2k) [inline scripts/uninstall]: Ha! you know i remember wondering why grafana was installed on time after i uninstalled it.... that would explain it.. 

Added uninstall script

File: scripts/grafana/install (hunk 3 above)
Comment (darkdrgn2k) [inline scripts/grafana/install]: Yes be but i want to make sure the database is started and initialized which happens upon daemon start to shoe horn the data source in there..

i'm not sure if it forked before or after the creation of the database

**Pull request merged**